docs/dev/design/new-search-api.rst

```diff
@@ -0,0 +1,297 @@
+
+CORS
+----
+
+Since the request could be attached to more than one project.
+we can't make the decision if we should enable CORS or not on a given request from the middleware easily,
+so we won't allow cross site requests when using the new API for now.
+We would need to refactor our CORS code,
+so every view can decide if CORS should be allowed or not,
+for this case, cross site requests will be allowed only if all versions of the final search are public,
+another alternative could be to always allow cross site requests,
+but when a request is cross site, we only return results from public versions.
+
+Analytics
+---------
+
+We will record the same query for each project that was used in the final search.
+
+Response
+--------
+
+The response will be similar to the old one,
+but will include extra information about the search,
+like the projects, versions, and the query that were used in the final search.
+
+And the ``version``, ``project``, and ``project_alias`` attributes will
+now be objects.
+
+We could just re-use the old response too,
+since the only breaking changes would be the attributes now being objects,
+and we aren't adding any new information to those objects (yet).
+But also, re-using the current serializers shouldn't be a problem either.
+
+.. code-block:: json
+
+   {
+     "count": 1,
+     "next": null,
+     "previous": null,
+     "projects": [
+       {
+         "slug": "docs",
+         "versions": [
+           {
+             "slug": "latest"
+           }
+         ]
+       }
+     ],
+     "query": "The final query used in the search",
+     "results": [
+       {
+         "type": "page",
+         "project": {
+           "slug": "docs",
+           "alias": null
+         },
+         "version": {
+           "slug": "latest"
+         },
+         "title": "Main Features",
+         "path": "/en/latest/features.html",
+         "domain": "https://docs.readthedocs.io",
+         "highlights": {
+           "title": []
+         },
+         "blocks": [
+           {
+             "type": "section",
+             "id": "full-text-search",
+             "title": "Full-Text Search",
+             "content": "We provide search across all the projects that we host. This actually comes in two different search experiences: dashboard search on the Read the Docs dashboard and in-doc search on documentation sites, using your own theme and our search results. We offer a number of search features: Search across subprojects Search results land on the exact content you were looking for Search across projects you have access to (available on Read the Docs for Business) A full range of search operators including exact matching and excluding phrases. Learn more about Server Side Search.",
+             "highlights": {
+               "title": [
+                 "Full-<span>Text</span> Search"
+               ],
+               "content": []
+             }
+           },
+           {
+             "type": "domain",
```

Comment:
We could also maybe just drop domains from the new API (https://github.com/readthedocs/readthedocs.org/issues/9571), but that decision can be made later, since we would just stop returning these type of blocks.